### PR TITLE
docs: Initial README

### LINKED ISSUE #4: Create a README
* [x] describe the reasoning behind the NEAR Lake framework
* [x] show how to use
* [x] how to configure in order to achieve your goals
* [x] plans

### PR DESCRIPTION
**Pull request by khorolets**
@frol I was thinking of adding some sort of roadmap or at least plans in the README, but I'm not sure we have any further plans for the NEAR Lake Framework. I hope this project will be updated as rarely as possible. I'd appreciate your comment on that.

Closes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+```
+
+`~/.aws/config`
+```
+[default]
+region=us-west-2
+output=json
+```
```
Comment (frol) [inline README.md:70]: Does the configuration has to be in those files? Can we / should we consolidate the configuration into explicitly passed parameters (`LakeConfig`)?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        streamer_message.shards.len()
+    );
+}
+```
+
+## How to use
+
+### AWS S3 Credentials
+
+In order to be able to get objects from the AWS S3 bucket you need to provide the AWS credentials.
```
Comment (frol) [inline README.md:54]: Is there a direct link to the page where it is described how to get those credentials?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+## Configuration
+
+Everything should be configured before the start of your indexer application via `LakeConfig` struct.
+
+Available parameters:
+
+* `bucket: String` - provide the AWS S3 bucket name (`near-lake-testnet`, `near-lake-mainnet` or yours if you run your own NEAR Lake)
+* `region: String` - provide the region for AWS S3 bucket
+* `start_block_height: Option<u64>` - optional block height to start stream from, if `None` will start from the earliest available block on AWS S3 bucket (usualy, the genesis block)
+* `tracked_shards: Vec<u8>` - the list of indexes of the shards to track, if empty `Vec` is provided assumes you track all shards
```
Comment (frol) [inline README.md:97]: Update these after #6

File: README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+use near_lake_framework::LakeConfig;
+
+#[tokio::main]
+async fn main() -> Result<(), tokio::io::Error> {
+    // create a NEAR Lake Framework config
+    let config = LakeConfig {
+        bucket: "near-lake-testnet".to_string(), // AWS S3 bucket name
+        region: "eu-central-1".to_string(), // AWS S3 bucket region
+        start_block_height: Some(82422587), // the latest block height we've got from explorer.near.org for testnet
+        tracked_shards: vec![0, 1, 2, 3], // track all 4 existing shards
```
Comment (frol) [inline README.md:20]: Update these after #6

**Comment by frol** [comment]
> I was thinking of adding some sort of roadmap or at least plans in the README

Maybe we can just point out to 1.0 milestone: https://github.com/near/near-lake-framework/milestone/2

**Review by frol — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (khorolets) [reply README.md:70]: I haven't found a way to pass it explicitly, I guess AWS works like that. You need to have either files or env vars

Comment (frol) [reply README.md:70]: @khorolets That is awful API design 😞

File: README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+* `s3_bucket_name: String` - provide the AWS S3 bucket name (`near-lake-testnet`, `near-lake-mainnet` or yours if you run your own NEAR Lake)
+* `s3_region_name: String` - provide the region for AWS S3 bucket
+* `start_block_height: u64` - block height to start the stream from
+
+
+## Future plans
+
+We use Milestones with clearly defined acceptance criteria:
+
+* [ ] [MVP](https://github.com/near/near-lake-framework/milestone/1)
```
Comment (khorolets) [inline README.md:98]:
```suggestion
* [x] [MVP](https://github.com/near/near-lake-framework/milestone/1)
```

By merging this PR we are closing the MVP milestone

**Review by frol — APPROVED** [review]

**Pull request merged**